File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/component/JwtEntryPoint.java
package com.windea.demo.cloudcollect.core.component;

import lombok.extern.apachecommons.CommonsLog;
import org.springframework.http.HttpStatus;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * Jwt安全验证进入点。
 */
@Component
@CommonsLog
public class JwtEntryPoint implements AuthenticationEntryPoint {
	@Override
	public void commence(HttpServletRequest request, HttpServletResponse response, AuthenticationException e)
	throws IOException {
		response.sendError(HttpStatus.UNAUTHORIZED.value(), HttpStatus.UNAUTHORIZED.getReasonPhrase());
		log.error("Unauthorized error. Message: " + e.getMessage());
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/component/JwtEntryPoint.kt
package com.windea.demo.cloudcollect.core.component

import org.springframework.http.*
import org.springframework.security.core.*
import org.springframework.security.web.*
import org.springframework.stereotype.*
import javax.servlet.http.*

@Component
class JwtEntryPoint : AuthenticationEntryPoint {
	override fun commence(request: HttpServletRequest, response: HttpServletResponse,
		authException: AuthenticationException) {
		println("Unauthorized error. ${authException.message}")
		response.sendError(HttpStatus.UNAUTHORIZED.value(), HttpStatus.UNAUTHORIZED.reasonPhrase)
	}
}
